### PR TITLE
This allows specifying context as arguments

### LINKED ISSUE #389: Idea: have a way to specify context via command line
Something like repeat arguments:
```
cookiecutter mytemplate -Cname=my-project -Cgithub-user=ionelmc
```

Or maybe the whole json?
```
cookiecutter mytemplate --context='{"name": "my-project", "github-user": "ionelmc"}'
```

Or variable arguments?
```
cookiecutter mytemplate --context-name=my-project --context-github-user=ionelmc
```

### PR DESCRIPTION
**Pull request by michaeljoseph**
Master rebased #400.
Fixes #389.

Allows specifying or overriding variables by adding `key=value` arguments after the template.

E.g.:

    [marca@marca-mac2 test-cookiecutter-supervisor]$ cookiecutter --no-input https://github.com/msabramo/cookiecutter-supervisor.git program_name=foobar
    Cloning into 'cookiecutter-supervisor'...
    remote: Counting objects: 16, done.
    remote: Compressing objects: 100% (13/13), done.
    remote: Total 16 (delta 5), reused 8 (delta 2), pack-reused 0
    Unpacking objects: 100% (16/16), done.
    Checking connectivity... done.

    [marca@marca-mac2 test-cookiecutter-supervisor]$ cat foobar/supervisor.conf
    [program:foobar]
    numprocs                 = 1
    process_name             = %(program_name)s%(process_num)02d
    user                     = foobaruser
    command                  = gunicorn --bind 0.0.0.0:$PORT --paste development.ini
    startsecs                = 30
    priority                 = 12
    redirect_stderr          = true
    autorestart              = true
    autostart                = true
    stopwaitsecs             = 10
    stopsignal               = QUIT
    directory                = /opt/webapp/foobar
    stdout_logfile           = /var/log/sm/foobar-app-out.log
    stdout_logfile_maxbytes  = 10MB
    stdout_logfile_backups   = 0
    stdout_capture_maxbytes  = 1MB
    stderr_logfile           = /var/log/sm/foobar-app-err.log
    stderr_logfile_maxbytes  = 10MB
    stderr_logfile_backups   = 0
    stderr_capture_maxbytes  = 1MB
    environment              = NEW_RELIC_CONFIG_FILE="/opt/webapp/foobar/etc/newrelic.ini"

### CODE REVIEW

File: cookiecutter/cli.py [hunk 1]
```diff
@@ -89,6 +101,13 @@ def main(template, no_input, checkout, verbose, replay, overwrite_if_exists,
             level=logging.INFO
         )
 
+    if extra_context:
+        # Convert tuple -- e.g.: (u'program_name=foobar', u'startsecs=66')
+        # to dict -- e.g.: {'program_name': 'foobar', 'startsecs': '66'}
+        extra_context = dict(s.split('=', 1) for s in extra_context)
```
Comment (hackebrot) [inline cookiecutter/cli.py]: Let's move this to the click callback 😄

**Comment by michaeljoseph** [comment]
:bell:

File: README.rst [hunk 2]
```diff
@@ -136,6 +136,12 @@ Did someone say features?
 * Default context: specify key/value pairs that you want used as defaults
   whenever you generate a project
 
+* Inject extra context with command-line arguments:
+
+    .. code-block:: bash
+
+        $ cookiecutter --no-input https://github.com/msabramo/cookiecutter-supervisor.git program_name=foobar startsecs=10
```
Comment (hackebrot) [inline README.rst]: Let's shorten this for the sake of readability:

```bash
$ cookiecutter --no-input gh:msabramo/cookiecutter-supervisor program_name=foobar startsecs=10
```

**Comment by michaeljoseph** [comment]
:bell:

**Comment by pydanny** [comment]
👍

**Comment by msabramo** [comment]
Cool, thanks @michaeljoseph for rebasing for me!

**Pull request merged**